### PR TITLE
Ftr: dubbo context sample

### PR DESCRIPTION
**Pull request by cjphaha**
* 添加 dubbo context 样例
* 配置 .run 目录下 .xml 配置文件的环境变量和文件路径
* 添加集成测试

### CODE REVIEW

**Review by zhaoyunxing92 — APPROVED** [review]

File: context/dubbo/go-client/cmd/client.go [hunk 1]
```diff
@@ -20,54 +20,57 @@ package main
...
 	hessian "github.com/apache/dubbo-go-hessian2"
-
 	"github.com/dubbogo/gost/log"
 )
 
-import (
-	pkg2 "github.com/apache/dubbo-go-samples/context/dubbo/go-client/pkg"
-)
+type UserProvider struct {
+	GetContext func(ctx context.Context, req []interface{}) (rsp *ContextContent, err error)
```
Comment (Mulavar) [inline context/dubbo/go-client/cmd/client.go:34]: 尽量不要用[]interface{} 使用具体的类型

File: context/dubbo/go-client/cmd/client.go [hunk 2]
```diff
@@ -20,54 +20,57 @@ package main
...
 	gxlog.CInfo("\n\n\nstart to test dubbo")
-	rspContent := &pkg2.ContextContent{}
+
 	atta := make(map[string]interface{})
 	atta["string-value"] = "string-demo"
 	atta["int-value"] = 1231242
-	atta["user-defined-value"] = pkg2.ContextContent{InterfaceName: "test.interface.name"}
+	atta["user-defined-value"] = ContextContent{InterfaceName: "test.interface.name"}
 	reqContext := context.WithValue(context.Background(), constant.DubboCtxKey("attachment"), atta)
-	err := userProvider.GetContext(reqContext, []interface{}{"A001"}, rspContent)
```
Comment (Mulavar) [inline context/dubbo/go-client/cmd/client.go:66]: 同上

File: context/dubbo/go-server/conf/dubbogo.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
+dubbo:
+  registries:
+    "demoZK":
```
Comment (Mulavar) [inline context/dubbo/go-server/conf/dubbogo.yml:3]: 去掉引号

Comment (cjphaha) [reply context/dubbo/go-server/conf/dubbogo.yml]: fixed

**Review by LaurenceLiZhixin — APPROVED** [review]

**Comment by LaurenceLiZhixin** [comment]
merge config-enhance and restart again

File: context/dubbo/go-client/cmd/client.go [hunk 4]
```diff
@@ -20,50 +20,53 @@ package main
...
+	UserDefinedStrVal string
+	CtxStrVal         string
+	CtxIntVal         int64
+}
 
-func init() {
-	config.SetConsumerService(userProvider)
-	hessian.RegisterPOJO(&pkg2.ContextContent{})
+func (*ContextContent) JavaClassName() string {
+	return "org.apache.dubbo.User"
```
Comment (xavier-niu) [inline context/dubbo/go-client/cmd/client.go:53]: Is the type of ContextContent `org.apache.dubbo.User`?

File: context/dubbo/go-client/cmd/client.go [hunk 5]
```diff
@@ -20,50 +20,53 @@ package main
...
 }
 
 // need to setup environment variable "CONF_CONSUMER_FILE_PATH" to "conf/client.yml" before run
```
Comment (xavier-niu) [inline context/dubbo/go-client/cmd/client.go:56]: Please update comments for how to set up environment variable.

File: context/dubbo/go-server/cmd/server.go [hunk 6]
```diff
@@ -26,30 +27,68 @@ import (
...
+
+func (u *UserProvider) Reference() string {
+	return "userProvider"
+}
+
+func (u ContextContent) JavaClassName() string {
+	return "org.apache.dubbo.User"
+}
+
 // need to setup environment variable "CONF_PROVIDER_FILE_PATH" to "conf/server.yml" before run
```
Comment (xavier-niu) [inline context/dubbo/go-server/cmd/server.go:90]: same above

File: context/dubbo/go-server/cmd/server.go [hunk 7]
```diff
@@ -26,30 +27,68 @@ import (
...
+	gxlog.CInfo("rsp:%#v", rsp)
+	return &rsp, nil
+}
+
+func (u *UserProvider) Reference() string {
+	return "userProvider"
+}
+
+func (u ContextContent) JavaClassName() string {
+	return "org.apache.dubbo.User"
```
Comment (xavier-niu) [inline context/dubbo/go-server/cmd/server.go:87]: same above

File: context/dubbo/go-client/conf/dubbogo.yml [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+    demoZK:
+      protocol: zookeeper
+      timeout: 3s
+      address: 127.0.0.1:2181
+  protocols:
+    dubbo:
+      name: dubbo
+      port: 20000
+      # 在 params 中定义当前使用的协议特有的网络配置
+      # 如该 sample 配置的是 dubbo 协议（底层使用 getty 通信库）的参数
```
Comment (xavier-niu) [inline context/dubbo/go-client/conf/dubbogo.yml:12]: Use english please.

File: context/dubbo/go-server/conf/dubbogo.yml [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+    demoZK:
+      protocol: zookeeper
+      timeout: 3s
+      address: 127.0.0.1:2181
+  protocols:
+    dubbo:
+      name: dubbo
+      port: 20000
+      # 在 params 中定义当前使用的协议特有的网络配置
+      # 如该 sample 配置的是 dubbo 协议（底层使用 getty 通信库）
```
Comment (xavier-niu) [inline context/dubbo/go-server/conf/dubbogo.yml:12]: same above

**Pull request merged**